src/btree/traversal.ts

```diff
@@ -28,6 +28,11 @@ export class TraversalIterator {
     const root = rootResponse.rootNode;
     const offset = rootResponse.pointer;
     this.records = await this.tree.traverse(this.key, root, offset);
```

Comment:
instead of directly updating the recursive `traverse()` logic, we can directly catch here